std/typecons.d

```diff
@@ -6850,24 +6850,32 @@ RefCounted!(T, RefCountedAutoInitialize.no) refCounted(T)(T val)
 {
     static struct File
     {
+        static size_t nDestroyed;
         string name;
         @disable this(this); // not copyable
-        ~this() { name = null; }
+        ~this() { name = null; ++nDestroyed; }
     }
 
+    assert(File.nDestroyed == 0);
     auto file = File("name");
     assert(file.name == "name");
     // file cannot be copied and has unique ownership
     static assert(!__traits(compiles, {auto file2 = file;}));
+    assert(File.nDestroyed == 0);
```

Comment:
Neither is this one.

Reply:
... but I have kept the other comparison against zero, as I feel it helps clarify the behaviour -- as per newly-added comments too.

I hope it is okay with you.  (-: